### ISSUE TITLE
GPU tests take way too long

### ISSUE THREAD

**Issue by navidcy** [CliMA/Oceananigans.jl#2139]
Some of the buildkite pipelines take way too long. Specifically the `time_stepping_2`, `regression` and `solver` tests.

I suggest we further split them?

**Comment by simone-silvestri** [comment]
It seems like there are some problems with the GPU tests, @navidcy and @glwagner do you have some idea on why tests are running for so long?

**Comment by glwagner** [comment]
The regression tests take a long time just because there are quite a few tests. We recently added many new hydrostatic regression tests (10-20? or more?) where there used to be just a handful. The solver tests have always been a bottleneck but I think there are a few new solvers being tested which probably increased the test length? I'm not sure why `time_stepping_2` would have increased (possibly that was always a bit slow).

I definitely agree that CI time is a problem, which is why we opened #1962 to move the CI over to Caltech's central cluster and to split tests into a "nightly" category (running every night rather than on every PR / commit), and a per-commit category. However, we haven't made much progress on that PR.
